### ISSUE TITLE
How to train Casanovo

### ISSUE THREAD

**Issue by wsnoble** [Noble-Lab/casanovo#278]
*N.B. Copied from an email.*

> To better understand Casanovo, I aim to replicate your work of training Casanovo using the new version of the nine-species dataset. I have already downloaded this dataset from https://doi.org/doi:10.25345/c52v2ck8j. In the 'Getting Started' section, the command to train Casanovo is: `casanovo train --validation_peak_path validation_spectra.mgf training_spectra.mgf`. However, I am unsure about the paths for 'validation_peak_path,' 'validation_spectra.mgf,' and 'training_spectra.mgf.' In the 'casanovo train --help' command, it mentions that 'TRAIN_PEAK_PATH' is also required. Therefore, I am very confused about how to train casanovo and what parameters I need to provide.

**Comment by bittremieux** [comment]
`validation_peak_path` is a required argument and thus does not need to be specified with a flag but is implicitly assumed. In the example from the Getting Started guide, this correspond to the file `"training_spectra.mgf"` at the end.

To clearly differentiate between the training data and the validation data, the `--validation_peak_path` flag is used. The validation file in the Getting Started example is `"validation_spectra.mgf"`.

**Comment by wsnoble** [comment]
From the Dongcao:

I have already downloaded this dataset (MSV000090982) from [https://doi.org/doi:10.25345/c52v2ck8j. ](https://urldefense.com/v3/__https://doi.org/doi:10.25345/c52v2ck8j.__;!!K-Hz7m0Vt54!hlg9KSZ4sovtNpFrVgKvou4a93gKxZBJsJ5BrELJoHjwGW2zmGG4r1tBh7qEpYkUbgjxLOkLUvE$)How should I proceed to train Casanova with my data? I'm not sure what to set for --validation_peak_path, validation_spectra.mgf, and training_spectra.mgf parameters. Also, how is the training set divided from the validation set?

**Comment by bittremieux** [comment]
I also got a separate email. My response:

> The 9-species dataset on MassIVE does not divide the data into training and testing splits. Instead, Casanovo was trained in a leave-one-species-out fashion. I.e. to reproduce the results, you would repeatedly train Casanovo on eight of the species and evaluate its performance on the left-out species.
>
> Within each training set, the training and validation set (to determine the best model during training) were created using a disjoint peptide sequences strategy.
>
> "--validation_peak_path" is not a path, but the command-line flag that indicates that the following value is the path to the validation spectra.
